From e031512b46642bf87390ba17877410fcbc89db04 Mon Sep 17 00:00:00 2001
From: Guy Harris <guy@alum.mit.edu>
Date: Wed, 25 Jun 2014 11:45:29 -0700
Subject: [PATCH 11/15] Give more details for --time-stamp-precision.

(cherry picked from commit e76768c97a70934b8f3a41fe2df705c61a924a47)

Conflicts:
        tcpdump.1.in

[msekleta: replaced .LP with .IP to fix indentation of paragraphs]
---
 tcpdump.1.in | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/tcpdump.1.in b/tcpdump.1.in
index 69c1984..8304a55 100644
--- a/tcpdump.1.in
+++ b/tcpdump.1.in
@@ -371,13 +371,23 @@ time stamp type cannot be set for the interface, no time stamp types are
 listed.
 .TP
 .BI \-\-time\-stamp\-precision= tstamp_precision
-.PD
-Set the time stamp precision for the capture to
-\fItstamp_precision\fP. Currently supported are microseconds and
-nanoseconds. Note that availability of high precision time stamps (nanoseconds)
-and their actual accuracy is platform and HW dependent. Also note that when
-writing captures to the savefile, distinct magic number is used to distinguish
-savefiles which contains time stamps in nanoseconds.
+When capturing, set the time stamp precision for the capture to
+\fItstamp_precision\fP.  Note that availability of high precision time
+stamps (nanoseconds) and their actual accuracy is platform and hardware
+dependent.  Also note that when writing captures made with nanosecond
+accuracy to a savefile, the time stamps are written with nanosecond
+resolution, and the file is written with a different magic number, to
+indicate that the time stamps are in seconds and nanoseconds; not all
+programs that read pcap savefiles will be able to read those captures.
+.IP
+When reading a savefile, convert time stamps to the precision specified
+by \fItimestamp_precision\fP, and display them with that resolution.  If
+the precision specified is less than the precision of time stamps in the
+file, the conversion will lose precision.
+.IP
+The supported values for \fItimestamp_precision\fP are \fBmicro\fP for
+microsecond resolution and \fBnano\fP for nanosecond resolution.  The
+default is microsecond resolution.
 .TP
 .B \-K
 Don't attempt to verify IP, TCP, or UDP checksums.  This is useful for
-- 
1.8.3.1

